internal/gatewayapi/runner/runner.go

```diff
@@ -212,124 +218,327 @@ func (r *Runner) subscribeAndTranslate(sub <-chan watchable.Snapshot[string, *re
+						aggregatedResult.SecurityPolicies = append(aggregatedResult.SecurityPolicies, aggregatedPolicy)
+					}
+
 				}
 				for _, envoyExtensionPolicy := range result.EnvoyExtensionPolicies {
-					key := utils.NamespacedName(envoyExtensionPolicy)
-					if len(envoyExtensionPolicy.Status.Ancestors) > 0 {
-						r.ProviderResources.EnvoyExtensionPolicyStatuses.Store(key, &envoyExtensionPolicy.Status)
-						envoyExtensionPolicyStatusCount++
+					var aggregatedPolicy *egv1a1.EnvoyExtensionPolicy
+					for _, ap := range aggregatedResult.EnvoyExtensionPolicies {
+						if utils.NamespacedName(envoyExtensionPolicy) == utils.NamespacedName(ap) {
+							aggregatedPolicy = ap
+							break
+						}
 					}
-					delete(keysToDelete.EnvoyExtensionPolicyStatus, key)
-					r.keyCache.EnvoyExtensionPolicyStatus[key] = true
-				}
-				for _, backend := range result.Backends {
-					key := utils.NamespacedName(backend)
-					if len(backend.Status.Conditions) > 0 {
-						r.ProviderResources.BackendStatuses.Store(key, &backend.Status)
-						backendStatusCount++
+					// If already found (because processed in an earlier gatewayclass), append its status ancestors.
+					// Otherwise, append it to the list of *Policies in the accumulated result.
+					if aggregatedPolicy != nil {
+						aggregatedPolicy.Status.Ancestors = append(aggregatedPolicy.Status.Ancestors, envoyExtensionPolicy.Status.Ancestors...)
+					} else {
+						aggregatedPolicy = envoyExtensionPolicy
+						aggregatedResult.EnvoyExtensionPolicies = append(aggregatedResult.EnvoyExtensionPolicies, aggregatedPolicy)
 					}
-					delete(keysToDelete.BackendStatus, key)
-					r.keyCache.BackendStatus[key] = true
 				}
+				aggregatedResult.Backends = append(aggregatedResult.Backends, result.Backends...)
 				for _, extServerPolicy := range result.ExtensionServerPolicies {
-					key := message.NamespacedNameAndGVK{
-						NamespacedName:   utils.NamespacedName(&extServerPolicy),
-						GroupVersionKind: extServerPolicy.GroupVersionKind(),
+					var aggregatedPolicy *unstructured.Unstructured
+					for _, ap := range aggregatedResult.ExtensionServerPolicies {
+						if utils.NamespacedName(&extServerPolicy) == utils.NamespacedName(&ap) {
+							aggregatedPolicy = &ap
+							break
+						}
 					}
-					if statusObj, hasStatus := extServerPolicy.Object["status"]; hasStatus && statusObj != nil {
-						if statusMap, ok := statusObj.(map[string]any); ok && len(statusMap) > 0 {
-							policyStatus := unstructuredToPolicyStatus(statusMap)
-							r.ProviderResources.ExtensionPolicyStatuses.Store(key, &policyStatus)
-							extensionServerPolicyStatusCount++
+
+					// If already found (because processed in an earlier gatewayclass), append its status ancestors.
+					// Otherwise, append it to the list of *Policies in the accumulated result.
+					if aggregatedPolicy != nil {
+						ancestorsObject, found, err := unstructured.NestedFieldCopy(aggregatedPolicy.Object, "status", "ancestors")
```

Comment:
Nit: could we create a method to merge the status for `ExtensionServerPolicies`, and add a test for this method to verify it works as expected?